File: JavaScript/1653. Minimum Deletions to Make String Balanced.js
/**
 * @param {string} s
 * @return {number}
 */
var minimumDeletions = function(s) {
    var stk = [];
    var res=0;
        for(var i=0;i<s.length;i++){
        if(s[i] == 'a'){
            if(stk.length!=0){
                stk.pop();
                res++;
            }
        }
        else{
           stk.push(s[i]);
        }
    }
    return res;
};

File: Java/1653. Minimum Deletions to Make String Balanced.java
class Solution {
    public int minimumDeletions(String s) {
        Stack<Character> stk = new Stack<Character>();
        int res=0;
        for(int i=0;i<s.length();i++){
            if(s.charAt(i) == 'a'){
                if(stk.size()!=0){
                    stk.pop();
                    res++;
                }
            }
            else{
               stk.push(s.charAt(i));
            }
        }
        return res;
    }
}
